### PR TITLE
basket cleanup for petitions

### PR DESCRIPTION
**Pull request by Pomax**
This removes the client-side newsletter signup for petitions, and closes https://github.com/mozilla/foundation.mozilla.org/issues/1627

### CODE REVIEW

**Comment by Pomax** [comment]
@alanmoo this simply removes the basket signup for new petitions, but also turns off client-side signup for old (now obsolete?) petitions. Are we okay with that are there strong reasons to keep newsletter signup for no longer relevant petitions working?

**Comment by alanmoo** [comment]
No, I'm not comfortable with turning those off until we've migrated the existing petitions to CRM campaigns.

File: source/js/components/petition/petition.jsx [hunk 1]
```diff
@@ -261,31 +259,6 @@ export default class Petition extends React.Component {
     });
   }
 
-  /**
-   * sign the user up for the mozilla newsletter.
-   *
-   * @returns {promise} the result the XHR post attempt.
-   */
-  signUpToBasket() {
-    this.setState({ basketSubmitted: true });
```
Comment (ScottDowne) [inline source/js/components/petition/petition.jsx]: I think `basketSubmitted` can be removed from other parts of the code too?

**Comment by ScottDowne** [comment]
@Pomax just one potential comment?

Other than that the code looks fine, I'll let you and Alan worry about his comment.

File: source/js/components/petition/petition.jsx (hunk 1 above)
Comment (Pomax) [reply source/js/components/petition/petition.jsx]: good point

**Comment by Pomax** [comment]
actually landed in revised form as part of the petition refactor already

**Pull request closed without merging**